R/pkg/inst/tests/test_sparkSQL.R

```diff
@@ -52,9 +52,13 @@ test_that("infer types", {
                list(type = 'array', elementType = "integer", containsNull = TRUE))
   expect_equal(infer_type(list(1L, 2L)),
                list(type = 'array', elementType = "integer", containsNull = TRUE))
-  expect_equal(infer_type(list(a = 1L, b = "2")),
-               structType(structField(x = "a", type = "integer", nullable = TRUE),
-                          structField(x = "b", type = "string", nullable = TRUE)))
+  testStruct <- infer_type(list(a = 1L, b = "2"))
+  expect_true(class(testStruct) == "structType")
+  expect_true(length(testStruct$fields()) == 2)
+  testSF <- testStruct$fields()[[1]]
```

Comment:
could we write a small util function that checks a struct field ? i.e we could have something like
```
checkStructField <- function(expected, actualName, actualType, actualNullable) {
    expect_equal(expected$name(), actualName)
    ...
}
```
Then we can check both `a` and `b` easily